docs/book/v3/plugin-managers.md

```diff
@@ -0,0 +1,197 @@
+familiar with this class, as it has been the implementation we have shipped from
+initial versions. The `HydratorPluginManager` extends the zend-servicemanager
+`AbstractPluginManager`, and has the following behaviors:
+
+- It will only return `Zend\Hydrator\HydratorInterface` instances.
+- It defines short-name aliases for all shipped hydrators (the class name minus
+  the namespace), in a variety of casing combinations.
+- All but the `DelegatingHydrator` are defined as invokable services (meaning
+  they can be instantiated without any constructor arguments).
+- The `DelegatingHydrator` is configured as a factory-based service, mapping to
+  the `Zend\Hydrator\DelegatingHydratorFactory`.
+- No services are shared; a new instance is created each time you call `get()`.
+
+### HydratorPluginManagerFactory
+
+`Zend\Hydrator\HydratorPluginManager` is mapped to the factory
+`Zend\Hydrator\HydratorPluginManagerFactory` when wired to the dependency
+injection container.
+
+The factory will look for the `config` service, and use the `hydrators`
+configuration key to seed it with additional services. This configuration key
+should map to an array that follows [standard zend-servicemanager configuration](https://docs.zendframework.com/zend-servicemanager/configuring-the-service-manager/).
+
+## StandaloneHydratorPluginManager
+
+`Zend\Hydrator\StandaloneHydratorPluginManager` provides an implementation that
+has no dependencies on other libraries. **It can only load the hydrators shipped
+with zend-hydrator**.
+
+### StandardHydratorPluginManagerFactory
+
+`Zend\Hydrator\StandardHydratorPluginManager` is mapped to the factory
+`Zend\Hydrator\StandardHydratorPluginManagerFactory` when wired to the dependency
+injection container.
+
+## HydratorManager alias
+
+`Zend\Hydrator\ConfigManager` defines an alias service, `HydratorManager`. That
+service will point to `Zend\Hydrator\HydratorPluginManager` if
+zend-servicemanager is installed, and `Zend\Hydrator\StandaloneHydratorPluginManager`
+otherwise.
+
+## Custom plugin managers
+
+If you do not want to use zend-servicemanager, but want a plugin manager that is
+customizable, or at least capable of loading the hydrators you have defined for
+your application, you should write a custom implementation of
+`Zend\Hydrator\HydratorPluginManagerInterface`, and wire it to the
+`HydratorManager` service, and/or one of the existing service names.
+
+As an example, if you want a configurable solution that uses factories, and want
+those factories capable of pulling application-level dependencies, you might do
+something like the following:
+
+```php
+namespace YourApplication;
+
+use Psr\Container\NotFoundExceptionInterface;
+use Psr\Container\ContainerInterface;
+use RuntimeException;
+use Zend\Hydrator\HydratorInterface;
+use Zend\Hydrator\HydratorPluginManagerInterface;
+use Zend\Hydrator\StandaloneHydratorPluginManager;
+
+class CustomHydratorPluginManager implements
```

Comment:
I think this section goes too deep in detail: describing what interface is needed should be sufficient, no?

Reply:
It may be too much detail for you or me, but having a concrete example to look at is really useful for junior and medior developers. I want to demonstrate a complete solution - even if it's something they would need to alter for their specific needs.

Reply:
This is code sitting in the docs, hidden in a block in a very edge case area: I'd simplify it to the bone to prevent that.